File: Prgrms_lv2_70129.java
import java.util.Arrays;

public class Prgrms_lv2_70129 { // 이진 변환 반복하기

    public int[] solution(String s) {
        int idx = 0;
        long zeroCnt = 0;
        while (true) {
            zeroCnt += s.chars().filter((c) -> c == '0').count();   // stream 활용
//            zeroCnt += s.length() - s.replace("0", "").length();    // 평범한 방식
            s = s.replace("0", "");
            s = Integer.toBinaryString(s.length());
            idx++;
            if (s.equals("1")) {
                break;
            }
        }
        return new int[]{idx, (int) zeroCnt};
    }

    public static void main(String[] args) {
        Prgrms_lv2_70129 problem = new Prgrms_lv2_70129();
        System.out.println(Arrays.toString(problem.solution("110010101001")));
        System.out.println(Arrays.toString(problem.solution("01110")));
        System.out.println(Arrays.toString(problem.solution("1111111")));
    }

}

File: Prgrms_lv2_70129.py
# 이진 변환 반복하기
def solution(s):
    idx = 0
    zeroCnt = 0
    while True:
        zeroCnt += s.count("0")
        s = s.replace("0", "")
        s = str(format(len(s), "b"))
        idx += 1
        if s == "1":
            break
    return [idx, zeroCnt]


if __name__ == '__main__':
    print(solution("110010101001"))
    print(solution("01110"))
    print(solution("1111111"))
